### PR TITLE
{tools}[GCCcore/11.3.0] ccache v4.6.3

### PR DESCRIPTION
**Pull request by iotaka**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/c/ccache/ccache-4.6.3-GCCcore-11.3.0.eb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+checksums = ['f46ba3706ad80c30d4d5874dee2bf9227a7fcd0ccaac31b51919a3053d84bd05']
+
+osdependencies = [('glibc-static', 'libc6-dev', 'glibc-devel')]
+
+builddependencies = [
+    ('binutils', '2.38'),
+    ('CMake', '3.23.1'),
+    ('Ninja', '1.10.2'),
+    ('zstd', '1.5.2', ),
+    ('pkg-config', '0.29.2'),
```
Comment (smoors) [inline easybuild/easyconfigs/c/ccache/ccache-4.6.3-GCCcore-11.3.0.eb:28]: `pkg-config` is no longer maintained, so from `2022a` onward we prefer to use `pkgconf` where possible

File: easybuild/easyconfigs/c/ccache/ccache-4.6.3-GCCcore-11.3.0.eb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+preconfigopts = 'LDFLAGS=" -fuse-ld=bfd"'
+configopts = '-DENABLE_DOCUMENTATION=OFF -DENABLE_IPO=ON -DZSTD_LIBRARY="$EBROOTZSTD/lib/libzstd.a" '
+# disable hunt for faster linker, since using ld.gold may fail (on CentOS 8, for example)
+configopts += '-DUSE_FASTER_LINKER=OFF'
+
+sanity_check_paths = {
+    'files': ['bin/ccache'],
+    'dirs': []
+}
+sanity_check_commands = ['ccache --help']
```
Comment (smoors) [inline easybuild/easyconfigs/c/ccache/ccache-4.6.3-GCCcore-11.3.0.eb:45]: why did you remove the second sanity check command here? see ccache-4.6.3.eb

```
sanity_check_commands = [
    'ccache --help',
    # Ensure that the binary does not depend on any shared library
    'ldd $(which ccache) | grep -q "not a dynamic executable"',
]
```

File: easybuild/easyconfigs/c/ccache/ccache-4.6.3-GCCcore-11.3.0.eb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+dependencies = [
+    ('hiredis', '1.0.2'),
+]
+
+# use BFD linker rather than default ld.gold (required on CentOS 8)
+preconfigopts = 'LDFLAGS=" -fuse-ld=bfd"'
+configopts = '-DENABLE_DOCUMENTATION=OFF -DENABLE_IPO=ON -DZSTD_LIBRARY="$EBROOTZSTD/lib/libzstd.a" '
+# disable hunt for faster linker, since using ld.gold may fail (on CentOS 8, for example)
+configopts += '-DUSE_FASTER_LINKER=OFF'
```
Comment (smoors) [inline easybuild/easyconfigs/c/ccache/ccache-4.6.3-GCCcore-11.3.0.eb:39]: there are some other configopts in ccache-4.6.3.eb that are missing here.
please add them back (or explain why they are no longer needed).

**Review by smoors — CHANGES_REQUESTED** [review]

**Comment by iotaka** [comment]
I did some tests
with `'-DCMAKE_EXE_LINKER_FLAGS="-static"'`,  not compiles
so also `'ldd $(which ccache) | grep -q "not a dynamic executable"'`  is not usable

File: easybuild/easyconfigs/c/ccache/ccache-4.6.3-GCCcore-11.3.0.eb [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+checksums = ['f46ba3706ad80c30d4d5874dee2bf9227a7fcd0ccaac31b51919a3053d84bd05']
+
+osdependencies = [('glibc-static', 'libc6-dev', 'glibc-devel')]
+
+builddependencies = [
+    ('binutils', '2.38'),
+    ('CMake', '3.23.1'),
+    ('Ninja', '1.10.2'),
+    ('zstd', '1.5.2', ),
+    ('pkgconfig', '1.5.5', '-python'),
```
Comment (smoors) [inline easybuild/easyconfigs/c/ccache/ccache-4.6.3-GCCcore-11.3.0.eb:28]: pkgconf is not the same as pkgconfig

```suggestion
    ('pkgconf', '1.8.0'),
```

**Comment by smoors** [comment]
> I did some tests with `'-DCMAKE_EXE_LINKER_FLAGS="-static"'`, not compiles so also `'ldd $(which ccache) | grep -q "not a dynamic executable"'` is not usable

I just tested it and it works for me. maybe you made a mistake somewhere?

**Review by smoors — CHANGES_REQUESTED** [review]

**Comment by smoors** [comment]
@boegelbot: please test @ generoso

**Review by smoors — APPROVED** [review]
lgtm

**Pull request merged**